x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/aggregations/metrics/AnalyticsPercentilesAggregatorFactory.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.analytics.aggregations.metrics;
+
+import org.elasticsearch.search.aggregations.metrics.PercentileRanksAggregationBuilder;
+import org.elasticsearch.search.aggregations.metrics.PercentilesAggregationBuilder;
+import org.elasticsearch.search.aggregations.metrics.PercentilesConfig;
+import org.elasticsearch.search.aggregations.metrics.PercentilesMethod;
+import org.elasticsearch.search.aggregations.support.ValuesSourceRegistry;
+import org.elasticsearch.xpack.analytics.aggregations.support.HistogramValuesSourceType;
+
+public class AnalyticsPercentilesAggregatorFactory {
+    public static void registerPercentilesAggregator(ValuesSourceRegistry valuesSourceRegistry) {
+        valuesSourceRegistry.register(PercentilesAggregationBuilder.NAME,
+            HistogramValuesSourceType.HISTOGRAM,
+            (AnalyticsPercentilesAggregatorSupplier) (name, valuesSource, context, parent, percents, percentilesConfig, keyed,
+                                                      formatter, pipelineAggregators, metaData) -> {
+
+                if (percentilesConfig.getMethod().equals(PercentilesMethod.TDIGEST)) {
```

Comment:
Unfortunately I think we can't get around this if/else situation like we did in Core.  We could "wrap" the core percentilesConfig with an xpack config... but internally it'd do essentially the same if/else to delegate to the appropriate agg.

I don't see us extending this code anywhere further so the extra wrapper didn't seem useful.  If we ever decide to build more xpack percentiles/ranks stuff it might make sense though.